kernel/generic/src/time/timeout.c

```diff
@@ -103,9 +88,6 @@ void timeout_register(timeout_t *timeout, uint64_t time,
 	irq_spinlock_lock(&CPU->timeoutlock, true);
 	irq_spinlock_lock(&timeout->lock, false);
 
-	if (timeout->cpu)
```

Comment:
If I understand this correctly, we are giving up the possibility to detect that a timeout is being registered for the second time here...